x/stakeibc/keeper/unbonding_records.go

```diff
@@ -315,69 +318,75 @@ func (k Keeper) SweepAllUnbondedTokensForHostZone(ctx sdk.Context, hostZone type
 			epochUnbondingRecordIds = append(epochUnbondingRecordIds, epochUnbondingRecord.EpochNumber)
 		}
 	}
-	// if we have any amount to sweep, then we can send the ICA call to sweep them
-	if totalAmtTransferToRedemptionAcct > 0 {
-		k.Logger(ctx).Info(fmt.Sprintf("\tSending batch SweepAllUnbondedTokens for %d amt to host zone %s", totalAmtTransferToRedemptionAcct, hostZone.ChainId))
-		// Issue ICA bank send from delegation account to redemption account
-		if (&hostZone).DelegationAccount != nil && (&hostZone).RedemptionAccount != nil { // only process host zones once withdrawal accounts are registered
-			// get the delegation account and rewards account
-			delegationAccount := hostZone.GetDelegationAccount()
-			if delegationAccount == nil || delegationAccount.Address == "" {
-				k.Logger(ctx).Error(fmt.Sprintf("Zone %s is missing a delegation address!", hostZone.ChainId))
-				return false, 0
-			}
-			redemptionAccount := hostZone.GetRedemptionAccount()
-			if redemptionAccount == nil || redemptionAccount.Address == "" {
-				k.Logger(ctx).Error(fmt.Sprintf("Zone %s is missing a redemption address!", hostZone.ChainId))
-				return false, 0
-			}
 
-			// build transfer message from delegation account to redemption account
-			sweepCoin := sdk.NewCoin(hostZone.HostDenom, sdk.NewInt(totalAmtTransferToRedemptionAcct))
-			var msgs []sdk.Msg
-			msgs = append(msgs, &banktypes.MsgSend{
-				FromAddress: delegationAccount.GetAddress(),
-				ToAddress:   redemptionAccount.GetAddress(),
-				Amount:      sdk.NewCoins(sweepCoin),
-			})
-			k.Logger(ctx).Info(fmt.Sprintf("Bank sending unbonded tokens batch, from delegation to redemption account. Msg: %v", msgs))
-
-			// store the epoch numbers in the callback to identify the epoch unbonding records
-			redemptionCallback := types.RedemptionCallback{
-				HostZoneId:              hostZone.ChainId,
-				EpochUnbondingRecordIds: epochUnbondingRecordIds,
-			}
+	// If we have any amount to sweep, then we can send the ICA call to sweep them
+	if totalAmtTransferToRedemptionAcct == 0 {
```

Comment:
Do we no longer need to chat that this is g.t. 0? It's an int64

Reply:
Good catch, I'll switch to <=0 

The idea was to move most of this out of the if statement (to avoid having the bulk of the function nested)
```
if amount > 0 {
   lots of code
} else {
   log something
}
```

its
```
if amount <= 0 {
  logs something
}

lots of code
```